cluster.go

```diff
@@ -125,6 +132,33 @@ func (cfg *ClusterConfig) CreateSession() (*Session, error) {
 	return NewSession(*cfg)
 }
 
+// translateHostPort is a helper method that will use the given AddressTranslator
+// if defined, to translate the given host:port (addr) into a new host:port string.
+// If no AddressTranslator or if an error occurs, the given host:port will be returned.
+func (cfg *ClusterConfig) translateHostPort(hostPort string) (string) {
+	if cfg.AddressTranslator == nil {
+		return hostPort
+	}
+	addr, port, err := parseIPPort(hostPort)
+	if err != nil {
+		return hostPort
+	}
+	newAddr, newPort := cfg.AddressTranslator.Translate(addr, port)
+	if gocqlDebug {
+		log.Printf("gocql: translating address '%s' to '%s:%d'", hostPort, newAddr.String(), newPort)
+	}
+	return net.JoinHostPort(newAddr.String(), strconv.Itoa(newPort))
+}
+
+func parseIPPort(ipPort string) (net.IP, int, error) {
+	addr, portStr, err := net.SplitHostPort(ipPort)
+	if err != nil {
+		return nil, 0, err
+	}
+	port, _ := strconv.Atoi(portStr)
+	return net.ParseIP(addr), port, nil
```

Comment:
I should point out that there is a similar problem with the port number. Use `net.LookupPort` instead of `Atoi`.

Lookup errors should either be surfaced or handled by returning the original addr and logging.